syllabus/core/stat_recorder.py

```diff
@@ -34,45 +31,96 @@ def record(self, episode_return: float, episode_length: int, episode_task, env_i
         """
```

Comment:
If you use defaultdicts for the stats you can cut out a lot of code. Change:
`self.stats = {task: {} for task in self.tasks}`
to:
`self.stats = {task: defaultdict(float) for task in self.tasks}`

Reply:
`from collections import defaultdict`